crates/wasmtime/src/trampoline/func.rs

```diff
@@ -210,7 +212,14 @@ pub fn create_handle_with_function(
 ) -> Result<(StoreInstanceHandle, VMTrampoline)> {
     let isa = {
         let isa_builder = native::builder();
-        let flag_builder = settings::builder();
+        let mut flag_builder = settings::builder();
+
+        if ft.params().iter().any(|p| *p == ValType::ExternRef)
+            || ft.results().iter().any(|r| *r == ValType::ExternRef)
+        {
+            flag_builder.enable("enable_safepoints").unwrap();
```

Comment:
Oh dear this is nasty, can we load flags from the `Store` instead of trying to recreate them here though?

Reply:
(in that I think we have a `Box<dyn TargetIsa>` shoved somewhere in there I think